pkg/artifacts/artifact_storage_test.go

```diff
@@ -165,6 +256,132 @@ func TestInitializeArtifactStorageWithConfigMap(t *testing.T) {
+				Name:      "pipelineruntest",
+			},
+		},
+		hasPVC: false,
+	}, {
+		desc: "location empty",
+		configMap: &corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: system.GetNamespace(),
+				Name:      v1alpha1.BucketConfigName,
+			},
+			Data: map[string]string{
+				v1alpha1.BucketLocationKey:              "",
+				v1alpha1.BucketServiceAccountSecretName: "secret1",
+				v1alpha1.BucketServiceAccountSecretKey:  "sakey",
+			},
+		},
+		pipelinerun: &v1alpha1.PipelineRun{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: "foo",
+				Name:      "pipelineruntest",
+			},
+		},
+		hasPVC: true,
+	}, {
+		desc: "missing location",
+		configMap: &corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: system.GetNamespace(),
+				Name:      v1alpha1.BucketConfigName,
+			},
+			Data: map[string]string{
+				v1alpha1.BucketServiceAccountSecretName: "secret1",
+				v1alpha1.BucketServiceAccountSecretKey:  "sakey",
+			},
+		},
+		pipelinerun: &v1alpha1.PipelineRun{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: "foo",
+				Name:      "pipelineruntest",
+			},
+		},
+		hasPVC: true,
+	}, {
+		desc: "no config map data",
+		configMap: &corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: system.GetNamespace(),
+				Name:      v1alpha1.BucketConfigName,
+			},
+		},
+		pipelinerun: &v1alpha1.PipelineRun{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: "foo",
+				Name:      "pipelineruntest",
+			},
+		},
+		hasPVC: true,
+	}, {
+		desc: "no secret",
+		configMap: &corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: system.GetNamespace(),
+				Name:      v1alpha1.BucketConfigName,
+			},
+			Data: map[string]string{
+				v1alpha1.BucketLocationKey: "gs://fake-bucket",
+			},
+		},
+		pipelinerun: &v1alpha1.PipelineRun{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: "foo",
+				Name:      "pipelineruntest",
+			},
+		},
+		hasPVC: false,
+	}} {
+		t.Run(c.desc, func(t *testing.T) {
+			fakekubeclient := fakek8s.NewSimpleClientset(c.configMap)
+
+			if c.hasPVC {
+				fakekubeclient = fakek8s.NewSimpleClientset(c.configMap, GetPVCSpec(c.pipelinerun))
+			}
+			if c.hasPVC {
+				_, err := fakekubeclient.CoreV1().PersistentVolumeClaims(c.pipelinerun.Namespace).Get(GetPVCName(c.pipelinerun), metav1.GetOptions{})
+				if err != nil {
+					t.Fatalf("Error getting expected PVC %s for PipelineRun %s: %s", GetPVCName(c.pipelinerun), c.pipelinerun.Name, err)
+				}
+			}
+			if err := CleanupArtifactStorage(c.pipelinerun, fakekubeclient, logger); err != nil {
+				t.Fatalf("Error cleaning up artifact storage: %s", err)
+			}
+			_, err := fakekubeclient.CoreV1().PersistentVolumeClaims(c.pipelinerun.Namespace).Get(GetPVCName(c.pipelinerun), metav1.GetOptions{})
+			if err != nil && !errors.IsNotFound(err) {
+				t.Fatalf("Error checking if PVC %s for PipelineRun %s has been cleaned up: %s", GetPVCName(c.pipelinerun), c.pipelinerun.Name, err)
+			}
+			if !errors.IsNotFound(err) {
```

Comment:
i think this block and the previous could be made a bit easier to read, e.g.:

```
if err == nil {
   t.Fatalf("expected error....")
} else if !errors.IsNotFound(err) {
   t.Fatalf("found PVC...")
}
```

Reply:
Okiedokie.